app/api/ada/adaTransactions/adaNewTransactions.js

```diff
@@ -114,42 +129,57 @@ export async function getAllUTXOsForAddresses(
   }
 }
 
+/** Perform the cryptography required to create a transaction */
 export async function getAdaTransactionFromSenders(
   senders: AdaAddresses,
   receiver: string,
   amount: string,
   cryptoWallet: CryptoWallet
 ): Promise<[SpendResponse, AdaAddress]> {
+  // Get all addresses in the single account to a list
   const cryptoAccount = getSingleCryptoAccount();
   const addressesMap = await getAdaAddressesMap();
   const addresses = mapToList(addressesMap);
+
+  // fetch new internal address from HD Wallet for change
   const changeAdaAddr = await createAdaAddress(cryptoAccount, addresses, 'Internal');
   const changeAddr = changeAdaAddr.cadId;
-  const outputs = [{ address: receiver, value: amount }];
-  const senderUtxos = await getAllUTXOsForAddresses(_getAddresses(senders));
+
+  // Get all user UTXOs
+  const senderUtxos = await getAllUTXOsForAddresses(_addressesToPublicHash(senders));
+
+  // Consider any UTXO as a possible input
   const inputs = _mapUTXOsToInputs(senderUtxos, addressesMap);
+  const outputs = [{ address: receiver, value: amount }];
+
+  // Note: selection policy is decided on the js-cardano-wasm side
   const result: SpendResponse = getResultOrFail(
     Wallet.spend(cryptoWallet, inputs, outputs, changeAddr)
   );
   return [result, changeAdaAddr];
 }
 
+/** Perform the cryptography required to create a transaction */
 async function _getAdaTransaction(
   receiver: string,
   amount: string,
   cryptoWallet: CryptoWallet,
-) {
+): Promise<[SpendResponse, AdaAddress]> {
+  // Consider all user addresses valid for the source of a transaction
```

Comment:
probably we could add a warning if some of the addresses are over the limit supported by our backend